core/engine/src/spanned_source_text.rs

```diff
@@ -0,0 +1,52 @@
+use boa_ast::{LinearSpan, SourceText};
+use boa_gc::{Finalize, Gc, Trace};
+
+#[derive(Trace, Finalize)]
+pub(crate) struct SourceTextInner {
+    #[unsafe_ignore_trace]
+    source_text: SourceText,
+}
+impl SourceTextInner {
+    #[must_use]
+    pub(crate) fn new(code: &mut boa_ast::Script) -> Option<Gc<Self>> {
+        code.take_source()
+            .map(|source_text| Gc::new(Self { source_text }))
+    }
+}
+
+/// Contains pointer to source code and span of the object.
+#[derive(Trace, Finalize, Clone)]
+pub struct SpannedSourceText {
+    gc: Gc<SourceTextInner>,
+    #[unsafe_ignore_trace]
+    span: LinearSpan,
+}
```

Comment:
What do you think about leaving out this struct completeley and having the two fields exist separetley where needed (`CodeBlock` and `Bytecompiler`)? On the `Bytecompiler` we already have the source text anyway and with this we basically duplicate it.